src/par_iter/internal.rs

```diff
@@ -7,14 +7,20 @@ use join;
 use super::IndexedParallelIterator;
 use super::len::*;
 
+pub trait ProducerContinuation<ITEM> {
```

Comment:
It is roughly what I mean. :) I could perhaps just use the term "callback", which is more accurate I guess and more common.